workbench-core/datasets/src/dataSetService.ts

```diff
@@ -850,13 +857,18 @@ export class DataSetService {
     dataset: DataSet,
     request: CreateProvisionDatasetRequest
   ): Promise<DataSetPermission[]> {
-    const permissions = request.permissions ?? [];
-    if (!permissions.some((p) => p.identity === request.authenticatedUser.id && p.identityType === 'USER')) {
-      permissions.push({
-        identity: request.authenticatedUser.id,
-        identityType: 'USER',
-        accessLevel: 'read-only'
-      });
+    let permissions: DataSetPermission[];
+
+    if (_.isEmpty(request.permissions)) {
+      permissions = [
+        {
+          identity: request.owner ? request.owner : request.authenticatedUser.id,
+          identityType: request.owner ? request.ownerType! : 'USER',
+          accessLevel: 'read-only'
+        }
+      ];
+    } else {
+      permissions = request.permissions!;
```

Comment:
```suggestion
    const permissions: DataSetPermission[] = request.permissions ?? [
        {
          identity: request.owner ? request.owner : request.authenticatedUser.id,
          identityType: request.owner ? request.ownerType! : 'USER',
          accessLevel: 'read-only'
        }
      ];
```

Reply:
this fails if request.permissions === []

Reply:
```suggestion
    const permissions: DataSetPermission[] = request.permissions?.length ? request.permissions : [
        {
          identity: request.owner ? request.owner : request.authenticatedUser.id,
          identityType: request.owner ? request.ownerType! : 'USER',
          accessLevel: 'read-only'
        }
      ];
```